scio-hdfs/src/main/scala/com/spotify/scio/hdfs/package.scala

```diff
@@ -99,6 +118,61 @@ package object hdfs {
         .map(_.getKey.datum())
     }
 
+    /**
+     * Hadoop source flavoured distributed cache - any source supported by Hadoop Filesystem
+     * abstraction can be used as a source for distributed cache.
+     *
+     * @param path to the artifact - it can be local/HDFS/S3/GCS ...
+     * @param conf optional custom Hadoop configuration
+     * @param username optional Hadoop Simple Authentication remote username
+     */
+    def hadoopDistCache[F](path: String,
+                           conf: Configuration = null,
+                           username: String = null)
+                          (initFn: File => F): DistCache[F] = self.pipelineOp {
+      if (self.isTest) {
+        //TODO: if test - can the path be remote HDFS/S3 etc?
+        self.distCache(path)(initFn)
+      } else {
```

Comment:
This `else` block basically assumes the file is on HDFS, so it doesn't really support S3/local/GCS unless they are configured in Hadoop conf and the required jars are loaded?